software/py/nbf.py

```diff
@@ -233,7 +233,7 @@ def init_dmem(self, pod_origin_x, pod_origin_y):
         x_eff = self.tgo_x + x + pod_origin_x
         y_eff = self.tgo_y + y + pod_origin_y
           
-        for k in range(1024, self.bsg_data_end_addr):
+        for k in range(0, self.bsg_data_end_addr):
```

Comment:
I'm generally not a fan of hard-coded values. Is there not a self.bsg_data_start_addr that can be referenced?

`DMEM_BASE_EPA = 0x400` on line 23 should be updated too.